packages/editor-ui/src/modules/credentials.ts

```diff
@@ -170,14 +184,54 @@ const module: Module<ICredentialsState, IRootState> = {
 		},
 		createNewCredential: async (context: ActionContext<ICredentialsState, IRootState>, data: ICredentialsDecrypted) => {
 			const credential = await createNewCredential(context.rootGetters.getRestApiContext, data);
-			context.commit('upsertCredential', credential);
+
+			if (context.rootGetters['settings/isEnterpriseFeatureEnabled'](EnterpriseEditionFeature.Sharing)) {
+				if (data.ownedBy) {
+					credential.ownedBy = data.ownedBy;
+				}
+
+				if (data.sharedWith) {
+					credential.sharedWith = data.sharedWith;
+				}
```

Comment:
is this correct order? should this happen after `disptch('setCredentialSharedWith'` below instead? in case of error in sharing, would not this be wrong?

Reply:
Maybe this could've been improved a bit. I was doing this because the backend wasn't sending me the sharing info in the create and update endpoints.

Updated to be clearer and cleaner. Thanks!